src/librustc_borrowck/diagnostics.rs

```diff
@@ -286,6 +286,54 @@ You can read more about cell types in the API documentation:
 https://doc.rust-lang.org/std/cell/
 "##,
 
+E0389: r##"
+An attempt was made to assign to data in a non-mutable reference. Example:
```

Comment:
Not very clear. Also, you have to write "Example of erroneous code" and not just "Example" as stated by the [RFC](https://github.com/rust-lang/rfcs/pull/1567). It's to keep error explanations homogeneous.